### PR TITLE
Jira 1738 Dial Timeout does not cancel task when Callee is busy/Unava…

### PR DESCRIPTION
**Pull request by xhoaluu**
…ilable

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by xhoaluu** [comment]
org.restcomm.connect.testsuite.UssdPullTest.testUssdPull
org.restcomm.connect.testsuite.telephony.DialRecordingTest.testDialClientAlice_AliceDisconnects

These testcases are passed when running on local environment.

File: restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+import java.nio.file.StandardCopyOption;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author hoan.h.luu@telestax.com
+ */
+public class MockPlayDelayMediaGateway extends RestcommUntypedActor {
```
Comment (maria-farooq) [inline restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java]: @xhoaluu a couple of clarifications needed:
1. as per issue description it does not seem that there is any issue related to play delay then why are we adding a  `MockPlayDelayMediaGateway`
2. If `MockPlayDelayMediaGateway` is needed for some reason, it should inherit the existing `MockMediaGateway` and enhance only the specific part of MediaGateway that we want to mock. can you please share which part of this MG is enhanced and do it by inheritance.

File: restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/DialForkPlayDelayTest.java [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+import org.restcomm.connect.commons.annotations.FeatureExpTests;
+import org.restcomm.connect.commons.annotations.SequentialClassTests;
+import org.restcomm.connect.commons.annotations.UnstableTests;
+import org.restcomm.connect.testsuite.NetworkPortAssigner;
+import org.restcomm.connect.testsuite.WebArchiveUtil;
+import org.restcomm.connect.testsuite.http.RestcommCallsTool;
+import org.restcomm.connect.testsuite.tools.MonitoringServiceTool;
+
+/**
+ * Tests for the Dial forking
```
Comment (maria-farooq) [inline restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/DialForkPlayDelayTest.java]: can we update the test elaboration here on what kind of scenario this class is testing. a comment can help future developer as well as reviewers to help understand the use case which are covered under a test

File: restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java (hunk 1 above)
Comment (xhoaluu) [reply restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java]: @maria-farooq Thanks for comments

Issue is when Finish Dial verb having timeout(default is 30 seconds), If after Dial verb state of call is Playing, then timeout of dial verb will finish the call.

1/ As MediaMockGateway immediately send back NTFY after RQNT for playing an audio file, that mean I cannot create playing state for a while to wait Dial verb timeout occured. Then I need to delay 2 seconds between a RQNT and NTFY when RC request RMS playing cache file to make issue happened.

2/ I also want to inherit MockMediaGateway. but there are a lot of private functions to come to my edited code. For more easy to later modification, I copied whole MediaMockGateway to new class and added my modification.

https://github.com/RestComm/Restcomm-Connect/blob/aaa40256fd3451a01ef41edd11d6d5bb314d1a85/restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java#L663

File: restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/DialForkPlayDelayTest.java (hunk 2 above)
Comment (xhoaluu) [reply restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/DialForkPlayDelayTest.java]: You are right. I will update it together with the other comments above when we have the final result.

File: restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java (hunk 1 above)
Comment (maria-farooq) [reply restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java]: @xhoaluu you can change `notify` method from `private` to `protected` and then override it in new class

Comment (xhoaluu) [reply restcomm/restcomm.mgcp/src/main/java/org/restcomm/connect/mgcp/MockPlayDelayMediaGateway.java]: As I don't want to touch to legacy code, but If you said that, I can update it and avoid code duplicated.

**Comment by maria-farooq** [comment]
@xhoaluu once you have green light from ci on new changes, feel free to merge the PR, i have no further comments/concerns.
and thanks for your great contribution.

**Pull request merged**